### ISSUE TITLE
SDL 2 does not detect key-up for left shift

### ISSUE THREAD

**Issue by MichaelMacha** [emscripten-core/emscripten#12274]
More likely, it doesn't recognize it for any control keys; but I'll start with left shift because it's confirmed.

I'm working on a project through emscripten, and for desktop—both compilations occur in my make file—which keeps track of key-down's and key-ups through a keyboard status buffer. The specific relevant code is as follows.

`#include "input.h"

void doKeyUp(SDL_KeyboardEvent *event)
{
	if (event->repeat == 0 && event->keysym.scancode < MAX_KEYBOARD_KEYS)
	{
		app.keyboard[event->keysym.scancode] = 0;
	}
}

void doKeyDown(SDL_KeyboardEvent *event)
{
	if (event->repeat == 0 && event->keysym.scancode < MAX_KEYBOARD_KEYS)
	{
		app.keyboard[event->keysym.scancode] = 1;
	}
}

void doInput(void)
{
	SDL_Event event;
	
	memset(app.inputText, '\0', MAX_LINE_LENGTH);
	
	while (SDL_PollEvent(&event))
	{
		switch (event.type)
		{
			case SDL_QUIT:
				exit(0);
				break;
				
			case SDL_KEYDOWN:
				doKeyDown(&event.key);
				break;
				
			case SDL_KEYUP:
				doKeyUp(&event.key);
				break;
				
			case SDL_TEXTINPUT:
				STRNCPY(app.inputText, event.text.text, MAX_LINE_LENGTH);
				break;

			default:
				break;
		}
	}
}`

One of the relevant keys I'm looking at is depressing SDLK_LSHIFT. In the desktop version (Linux Mint 19), it works fine; but in the emscripten version, pressing shift is detected, but it never lets up regardless of the status of my keyboard. The key is effectively held down until applet termination.

**Comment by MichaelMacha** [comment]
Update.

I've attempted to get around this issue by manually setting the keyboard buffer to reset after a single shot, which is probably what I should be doing anyway, and I found something interesting.

It only fails on Chrome (that I know of). On Firefox, it works just like on the desktop, though Chrome seems to handle emscripten much more efficiently.

It's as if shift is being intercepted by the browser. I'm going to try switching keys for specials and see if that helps; this may have something to do with some exotic Google API behavior.
